File: Beecrowd/1533 Detetive Watson/main.c
#include <stdio.h>

int obtemMaior(int *pVetor, int pTam);
int obtemIndiceSegundoMaior(int *pVetor, int pTam, int maior);

int main()
{
    int n, maior, indiceSegundoMaior;

    while (1)
    {
        scanf("%d", &n);

        if (n == 0)
            break;

        int vet[n];

        for (int i = 0; i < n; i++)
        {
            scanf("%d", &vet[i]);
        }

        maior = obtemMaior(vet, n);

        indiceSegundoMaior = obtemIndiceSegundoMaior(vet, n, maior) + 1;

        printf("%d\n", indiceSegundoMaior);
    }

    return 0;
}

int obtemMaior(int *pVetor, int pTam)
{
    int maior = pVetor[0];

    for (int i = 0; i < pTam; i++)
    {
        if (pVetor[i] > maior)
        {
            maior = pVetor[i];
        }
    }
    return maior;
}

int obtemIndiceSegundoMaior(int *pVetor, int pTam, int maior)
{
    int segundoMaior = pVetor[0];
    int cont = 0;
    if (pVetor[0] == maior)
    {
        segundoMaior = pVetor[1];
        cont = 1;
    }

    for (int i = 0; i < pTam; i++)
    {
        if (pVetor[i] > segundoMaior && pVetor[i] != maior)
        {
            segundoMaior = pVetor[i];
            cont = i;
        }
    }
    return cont;
}

File: Beecrowd/1533 Detetive Watson/Main.java
import java.io.IOException;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);

        int n;

        while (true) {
            n = sc.nextInt();

            if (n == 0)
                break;

            int[] vet = new int[n];

            for (int i = 0; i < n; i++)
                vet[i] = sc.nextInt();

            int maior = obtemMaior(vet);
            int indiceSegundoMaior = obtemIndiceSegundoMaior(vet, maior) + 1;

            System.out.println(indiceSegundoMaior);

        }

        sc.close();

    }

    public static int obtemMaior(int[] vet) {
        int maior = vet[0];

        for (int i = 0; i < vet.length; i++) {
            if (vet[i] > maior) {
                maior = vet[i];
            }
        }
        return maior;
    }

    public static int obtemIndiceSegundoMaior(int[] vet, int maior) {
        int segundoMaior = vet[0];
        int cont = 0;
        if (vet[0] == maior) {
            segundoMaior = vet[1];
            cont = 1;
        }

        for (int i = 0; i < vet.length; i++) {
            if (vet[i] > segundoMaior && vet[i] != maior) {
                segundoMaior = vet[i];
                cont = i;
            }
        }
        return cont;
    }

}
